python/lsst/ctrl/mpexec/dotTools.py

```diff
@@ -282,10 +281,8 @@ def expand_dimensions(connection: connectionTypes.BaseConnection) -> list[str]:
     allDatasets: set[str | tuple[str, str]] = set()
     if isinstance(pipeline, Pipeline):
         # TODO: DM-40639 will rewrite this code and finish off the deprecation
-        # of toExpandedPipeline.
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", category=FutureWarning)
-            pipeline = pipeline.toExpandedPipeline()
+        # of toExpandedPipeline but for now use the compatibility API.
+        pipeline = pipeline.to_graph()._iter_task_defs()
```

Comment:
@TallJimbo I assume this is okay for now (I got it from the old definition of `toExpandedPipeline`).

Reply:
Yes, `_iter_task_defs` will stick around until `TaskDef` goes away.